### PR TITLE
Register plugins with promises

### PR DESCRIPTION
**Pull request by cromefire**
**Changes**
Added the possibility to load plugins via Promises

### CODE REVIEW

File: src/components/pluginManager.js [hunk 1]
```diff
@@ -25,68 +25,85 @@ define(['events'], function (events) {
...
+                            pluginSpec = appRouter.baseUrl() + '/' + pluginSpec;
+                        }
+                    }
 
-                if (plugin.getRoutes) {
-                    plugin.getRoutes().forEach(function (route) {
-                        definePluginRoute(instance, route, plugin);
-                    });
-                }
+                    var separatorIndex = Math.max(pluginSpec.lastIndexOf('/'), pluginSpec.lastIndexOf('\\'));
```
Comment (JustAMan) [inline src/components/pluginManager.js:82]: why `lastIndexOf('\\')`? I mean why care for both slashes?

File: src/components/pluginManager.js [hunk 2]
```diff
@@ -25,68 +25,85 @@ define(['events'], function (events) {
...
+                });
+            }
+        }
 
-                if (existing) {
-                    resolve(url);
-                    return;
-                }
+        if (typeof pluginSpec === "string") {
+            console.debug('Loading plugin: ' + pluginSpec);
```
Comment (JustAMan) [inline src/components/pluginManager.js:57]: we should blurb that this is a deprecated method of registering a plugin here

Comment (cromefire) [reply src/components/pluginManager.js:57]: You mean like with a comment or so?

File: src/components/pluginManager.js (hunk 1 above)
Comment (cromefire) [reply src/components/pluginManager.js:82]: Well I did not add this 🤷🏻‍♀️

File: src/components/pluginManager.js (hunk 2 above)
Comment (JustAMan) [reply src/components/pluginManager.js:57]: no, I mean a message to browser console

File: src/components/pluginManager.js (hunk 1 above)
Comment (JustAMan) [reply src/components/pluginManager.js:82]: but you're the one who's changing the code right now ;)

Comment (cromefire) [reply src/components/pluginManager.js:82]: Well didn't want to break something so I left everything I didn't understand why it was there

File: src/components/pluginManager.js (hunk 2 above)
Comment (cromefire) [reply src/components/pluginManager.js:57]: Can do that

File: src/components/pluginManager.js (hunk 1 above)
Comment (JustAMan) [reply src/components/pluginManager.js:82]: Well, okay, let's keep backwards compat here.

File: src/components/pluginManager.js (hunk 2 above)
Comment (cromefire) [reply src/components/pluginManager.js]: Fixed

**Pull request merged**